nuget/spec/dependabot/nuget/update_checker/repository_finder_spec.rb

```diff
@@ -228,6 +228,36 @@
         )
       end
 
+      context "that include the default repository with disable external source" do
```

Comment:
💯 

We could also add a `.compact` to `disabled_sources` to remove the nil keys from disabled sources.

```
        def disabled_sources(doc)
          doc.css("configuration > disabledPackageSources > add").map do |node|
            value = node.attribute("value")&.value ||
                    node.at_xpath("./value")&.content
            if value&.strip&.downcase == "true"
              node.attribute("key")&.value&.strip ||
                node.at_xpath("./key")&.content&.strip
            end
          end.compact     <--
        end
```

Reply:
The problem is the key used as `nil`. If we use default source url's key as `nil` and compact in `disabled_sources` then the below code will never be remove the default resource. 
```
  sources.reject! do |s|
    disabled_sources.include?(s[:key])
  end
```

Reply:
After this change, we should update nuget ecosystem doc as well  that to disable default nuget registy without `<clear />`, user can add key 'nuget' to the `disabledPackageSources` in their nuget.config file.

Reply:
If we agreed to keep default registry key as `nuget` or `nuget.org` or some value then we will not need compact, though test case is still valid.